types/node/stream/promises.d.ts

```diff
@@ -0,0 +1,26 @@
+declare module "node:stream/promises" {
+    export * from "stream/promises";
+}
+
+declare module "stream/promises" {
+    import { FinishedOptions } from "stream";
+
+    function finished(stream: NodeJS.ReadableStream | NodeJS.WritableStream | NodeJS.ReadWriteStream, options?: FinishedOptions): Promise<void>;
+
+    function pipeline(stream1: NodeJS.ReadableStream, stream2: NodeJS.WritableStream): Promise<void>;
```

Comment:
You might want to bring in the changes from #52304 which improved the type inference for `pipeline`.

Reply:
I don't really understand the changes done here, while I generally agree I'd like to move this PR forward so we can get some extra versions in before node v16 lands.

@fluggo maybe you want to tackle this.

Reply:
I can look at it, but may be a day or two. The main thing would be to grab the `__promisify__` [definitions from stream.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/717af95f4120ea2b4a35e10df2ff2abdf923ed89/types/node/stream.d.ts#L392) and apply them to the stream/promise versions. Then later someone can add support for aborts, which were added in v15.